app/assets/javascripts/controllers/dialog_user/dialog_user_reconfigure_controller.js

```diff
@@ -0,0 +1,74 @@
+ManageIQ.angular.app.controller('dialogUserReconfigureController', ['API', 'dialogFieldRefreshService', 'miqService', 'resourceActionId', 'targetId', function(API, dialogFieldRefreshService, miqService, resourceActionId, targetId) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    var apiCall = new Promise(function(resolve) {
```

Comment:
isn't this whole thing just

```
API.get(url)
  .then(init)
  .then(miqService.refreshSelectpicker);
```

?

Reply:
Yeah I thought so too, but when I tried to use `.then(miqService.refreshSelectpicker)` in https://github.com/ManageIQ/manageiq-ui-classic/pull/4063, it did not work and had to manually resolve the promise first.